### PR TITLE
[DLIGHT][GPU] Enhance opencl thread limit for schedules

### PR DESCRIPTION
**Pull request by krishnaraj36**
Enhanced the opencl thread limit and improved the gpu schedules for opencl targets.
It improves decode performance 20 % for few set of MLC llm models.

LLM model --- baseline --- improved
gemma-2b-it --- 22.4 tok/sec --- 28.2 tok/sec
Qwen-7b-chat --- 11 tok/sec --- 11.8 tok/sec

### CODE REVIEW

**Comment by krishnaraj36** [comment]
@srkreddy1238  @tqchen : Please take a look to this PR and let me know your advise.

File: src/target/target_kind.cc [hunk 1]
```diff
@@ -340,9 +340,9 @@ TVM_REGISTER_TARGET_KIND("rocm", kDLROCM)
     .set_target_parser(UpdateROCmAttrs);
 
 TVM_REGISTER_TARGET_KIND("opencl", kDLOpenCL)
-    .add_attr_option<Integer>("max_threads_per_block", Integer(256))
+    .add_attr_option<Integer>("max_threads_per_block", Integer(1024))
     .add_attr_option<Integer>("max_shared_memory_per_block", Integer(16384))
-    .add_attr_option<Integer>("max_num_threads", Integer(256))
+    .add_attr_option<Integer>("max_num_threads", Integer(1024))
```
[the remark this answers is not in the document]
Comment (krishnaraj36) [reply src/target/target_kind.cc:345]: Thanks @tqchen !

This help in this PR  https://github.com/mlc-ai/mlc-llm/pull/2289 , which has great improvement in decode page attention kernel.

[the remark this answers is not in the document]
Comment (tqchen) [reply src/target/target_kind.cc:345]: OK, maybe we can take a different approach. When we do `--device android`, we can translate that to target opencl and overrides the `max_num_threads` to 1024

[the remark this answers is not in the document]
Comment (tqchen) [reply src/target/target_kind.cc:345]: see an example in https://github.com/mlc-ai/mlc-llm/blob/main/python/mlc_llm/support/auto_target.py#L424C11-L424C42

[the remark this answers is not in the document]
Comment (krishnaraj36) [reply src/target/target_kind.cc:345]: it make sense! We will do the update.

Thanks

**Comment by tqchen** [comment]
cc @mengshyu would be nice to confirm metal improvement and see if we want it for webgpu/metal

**Comment by mengshyu** [comment]
Hi @krishnaraj36 

After verification on Apple metal(m2), it seems that the performance has slightly declined, can you check it again on different platform, thanks.

- Baseline
commit:
    mlc llm:33c15e72a3567292cba577ea7f89652ec9f2bd6e
    relax:ced07e88781c0d6416e276d9cd084bb46aaf3da5

    -  model: gemma-2b-it
    Statistics:
        ----------- prefill -----------
        throughput: 554.084 tok/s
        total tokens: 7 tok
        total time: 0.013 s
        ------------ decode ------------
        throughput: 151.030 tok/s
        total tokens: 128 tok
        total time: 0.848 s


    - model: Phi-3-mini-4k-instruct
    Statistics:
        ----------- prefill -----------
        throughput: 350.196 tok/s
        total tokens: 7 tok
        total time: 0.020 s
        ------------ decode ------------
        throughput: 106.038 tok/s
        total tokens: 128 tok
        total time: 1.207 s




- Opt
commit:
    mlc llm:33c15e72a3567292cba577ea7f89652ec9f2bd6e
    relax commit:ced07e88781c0d6416e276d9cd084bb46aaf3da5 + patch

    - model: gemma-2b-it
    Statistics:
        ----------- prefill -----------
        throughput: 50.661 tok/s
        total tokens: 7 tok
        total time: 0.138 s
        ------------ decode ------------
        throughput: 149.744 tok/s
        total tokens: 128 tok
        total time: 0.855 s


    - model: Phi-3-mini-4k-instruct
    Statistics:
        ----------- prefill -----------
        throughput: 49.064 tok/s
        total tokens: 7 tok
        total time: 0.143 s
        ------------ decode ------------
        throughput: 104.042 tok/s
        total tokens: 128 tok
        total time: 1.230 s

**Pull request merged**